src/main/java/io/airbrake/javabrake/Notifier.java

```diff
@@ -156,4 +156,9 @@ Notice filterNotice(Notice notice) {
   public Notice buildNotice(Throwable e) {
     return new Notice(e);
   }
+
+  /** Get NoticeBuilder from an exception. */
+  public NoticeBuilder Builder(Throwable e) {
+    return new NoticeBuilder(e);
```

Comment:
Is it possible to pass the reference to the `Notifier` to the `NoticeBuilder` object?  If we do this, then it's possible to do something like this:

```java
try {
   // produce error
} catch (Exception e) {
   notifier.noticeBuilder(e)
     .environment("production")
     .context("severity", "backend")
     .param("key", "value")
     .send();
}
```

The `send()` method can perform the internal `build()` and then call the `notifier.send()` internally as well.

Just a thought...

Reply:
It might be good to have the `NoticeBuilder` exposed on the `Notice` class instead of the `Notifier` class.  This is a typical builder pattern approach.

Something like:

`Notice.java`:
```java
class Notice {

  public static Builder builder(final Notifier notifier, final Throwable exception ) {
    final Builder builder =  new Builder();
    builder.notifier = notifier;
    builder.exception = exception;
    return builder;
  }

  static class Builder {
   ...
    private Throwable exception;
    private Notifier notifier;

   public Notice build() { ... }

   public void send() {
      notifier.send(build());
   }
  }

}
```

This is probably more preferred than above, because it avoids any circular references that would occur otherwise.  With this design, we would remove the `noticeBuilder(Exception)` method from `Notifier` altogether.